fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java

```diff
@@ -694,8 +695,18 @@ public TExecPlanFragmentParams plan(TUniqueId loadId, long txnId) throws UserExc
             if (table == null) {
                 throw new MetaNotFoundException("table " + this.tableId + " does not exist");
             }
-            StreamLoadPlanner planner =
-                    new StreamLoadPlanner(db, (OlapTable) table, StreamLoadTask.fromRoutineLoadJob(this));
+            StreamLoadPlanner planner = new StreamLoadPlanner(db, (OlapTable) table, StreamLoadTask.fromRoutineLoadJob(this));
+            if (Config.enable_insert_planner_v2) {
+                long execMemLimit = 0;
+                if (getSessionVariables().containsKey(SessionVariable.EXEC_MEM_LIMIT)) {
+                    execMemLimit = Long.parseLong(getSessionVariables().get(SessionVariable.EXEC_MEM_LIMIT));
+                } else {
+                    execMemLimit = SessionVariable.DEFAULT_EXEC_MEM_LIMIT;
+                }
+                InsertPlannerV2 insertPlanner = new InsertPlannerV2(id, loadId, txnId, db.getId(), (OlapTable) table, 
+                        isStrictMode(), getTimezone(), isPartialUpdate(), ConnectContext.get(), sessionVariables, 
+                        execMemLimit, execMemLimit, false, getColumnDescs(), StreamLoadTask.fromRoutineLoadJob(this));
```

Comment:
Why to pass two execMemLimit in the parameter list?

Reply:
To be consistent with the behaviour of broker load